client/src/main/java/org/threadly/litesockets/client/http/HTTPClient.java

```diff
@@ -371,14 +401,53 @@ private TCPClient getTCPClient(final HTTPAddress ha) throws IOException {
   }
 
   private void addBackTCPClient(final HTTPAddress ha, final TCPClient client) {
+    if(maxIdleTime == 0) {
+      client.close();
+      return;
+    }
+    if(tcpClients.size() > this.maxConcurrent) {
+      limitConnections();
+    }
     if(!client.isClosed()) {
-      ArrayDeque<TCPClient> ll = sockets.get(ha);  
+      ArrayDeque<Pair<Long,TCPClient>> ll = sockets.get(ha);  
       if(ll == null) {
         sockets.put(ha, new ArrayDeque<>(8));
         ll = sockets.get(ha);
       }
       synchronized(ll) {
-        ll.add(client);
+        ll.add(new Pair<Long, TCPClient>(Clock.lastKnownForwardProgressingMillis(), client));
+      }
+    }
+  }
+  
+  private void limitConnections() {
+    if(tcpClients.size() > this.maxConcurrent) {
+      for(ArrayDeque<Pair<Long,TCPClient>> adq: sockets.values()) {
```

Comment:
I am pretty sure the logic is wrong here.  Wont this close all connections?  I assume the initial if condition needs to be checked below as well.